runtime/cudaq/platform/default/rest/RemoteRESTQPU.cpp

```diff
@@ -277,6 +289,35 @@ class RemoteRESTQPU : public cudaq::QPU {
     executor->setServerHelper(serverHelper.get());
   }
 
+  /// @brief Conditionally form an output_names JSON object if this was for QIR
+  nlohmann::json formOutputNames(const std::string &codegenTranslation,
+                                 const std::string &codeStr) {
+    // Form an output_names mapping from codeStr
+    nlohmann::json output_names;
+    std::vector<char> bitcode;
+    if (codegenTranslation == "qir") {
+      // decodeBase64 will throw a runtime exception if it fails
+      if (llvm::decodeBase64(codeStr, bitcode)) {
+        cudaq::info("Could not decode codeStr {}", codeStr);
+      } else {
+        llvm::LLVMContext llvmContext;
+        auto buffer = llvm::MemoryBuffer::getMemBufferCopy(
+            llvm::StringRef(bitcode.data(), bitcode.size()));
+        auto moduleOrError =
+            llvm::parseBitcodeFile(buffer->getMemBufferRef(), llvmContext);
+        if (moduleOrError.takeError())
+          throw std::runtime_error("Could not parse bitcode file");
+        auto module = std::move(moduleOrError.get());
+        for (llvm::Function &func : *module) {
+          if (func.hasFnAttribute("output_names"))
+            output_names = nlohmann::json::parse(
+                func.getFnAttribute("output_names").getValueAsString());
```

Comment:
Are we expecting only one function has this attribute? If so, we could early return/break the loop once we find the one.

Reply:
Yes, that was the intent, but I should probably double check that by adding a check for `"entry_point"`, too, and then I'll add the `break`. Thanks.

Reply:
Updated in 7926e58b.